### PR TITLE
Fix listAll functionality in Storage Emulator

### PR DESCRIPTION
**Pull request by tonyjhuang**
### Description

Fixes Firebase Rules integration from Storage emulator where the `delimiter` key was not being passed in the context correctly. This was causing all `list()` and `listAll()` invocations on the SDK to fail.

Also updated integration test setup for `list and listAll` methods to be more hermetic.

### Scenarios Tested

Tested via integration tests.

### CODE REVIEW

File: scripts/storage-emulator-integration/tests.ts [hunk 1]
```diff
@@ -1685,44 +1640,6 @@ describe("Storage emulator", () => {
...
-              customMetadata: {
-                testable: "true",
-              },
-            });
-        }, filename);
-
-        expect(metadata.contentType).to.equal("application/awesome-stream");
-        expect(metadata.customMetadata.testable).to.equal("true");
-      });
-
```
Comment (tonyjhuang) [inline scripts/storage-emulator-integration/tests.ts:1725]: moved to #updateMetadata below

File: src/emulator/storage/apis/firebase.ts [hunk 2]
```diff
@@ -147,10 +147,24 @@ export function createFirebaseEndpoints(emulator: StorageEmulator): Router {
...
+    // The prefix query param must be empty or end in a "/"
+    let prefix = "";
+    if (req.query.prefix) {
+      prefix = req.query.prefix.toString();
+      if (prefix.charAt(prefix.length - 1) !== "/") {
+        return res.status(400).json({
+          error: {
+            code: 400,
+            message:
+              "The prefix parameter is required to be empty or ends with a single / character.",
```
Comment (tonyjhuang) [inline src/emulator/storage/apis/firebase.ts:159]: This mirrors the Storage backend logic and error message.

File: scripts/storage-emulator-integration/tests.ts [hunk 3]
```diff
@@ -1260,19 +1243,23 @@ describe("Storage emulator", () => {
...
+      this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
+      await page.evaluate(async () => {
+        await firebase.auth().signOut();
+      });
       await page.close();
     });
 
-    after(async () => {
+    after(async function (this) {
+      this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
```
Comment (abhis3) [inline scripts/storage-emulator-integration/tests.ts:1255]: We set this in line 104, I think we can remove this. Unless we want the same timeout applied to Firebase Endpoints and the entire test suite?

Comment (tonyjhuang) [reply scripts/storage-emulator-integration/tests.ts:1255]: Done

**Review by abhis3 — APPROVED** [review]
LGTM thanks for all the changes!

**Review by Yuangwang — APPROVED** [review]
LGTM

**Pull request merged**